src/components/my-reading/saved-bookmarks.js

```diff
@@ -0,0 +1,312 @@
+import querystring from 'querystring'
+import PropTypes from 'prop-types'
+// @twreporters
+import { Title1 } from '@twreporter/react-components/lib/title-bar'
+import { CardList } from '@twreporter/react-components/lib/listing-page'
+import mq from '@twreporter/core/lib/utils/media-query'
+import twreporterRedux from '@twreporter/redux'
+import useBookmark from '@twreporter/react-components/lib/hook/use-bookmark'
+import useStore from '@twreporter/react-components/lib/hook/use-store'
+import {
+  SnackBar,
+  useSnackBar,
+} from '@twreporter/react-components/lib/snack-bar'
+import requestOrigin from '@twreporter/core/lib/constants/request-origins'
+import EmptyState from '@twreporter/react-components/lib/empty-state'
+import { Bookmark } from '@twreporter/react-components/lib/icon'
+import { P2 } from '@twreporter/react-components/lib/text/paragraph'
+// components
+import Pagination from '../Pagination'
+// context
+import { CoreContext } from '../../contexts'
+// constants
+import routes from '../../constants/routes'
+// lodash
+import get from 'lodash/get'
+import map from 'lodash/map'
+import filter from 'lodash/filter'
+
+const BOOKMARK_PER_PAGE = 10
+
+const _ = {
+  get,
+  map,
+  filter,
+}
+
+const Container = styled.div`
+  width: 100%;
+`
+const ListContainer = styled.div`
+  margin-top: 24px;
+  margin-bottom: 64px;
+  ${mq.mobileOnly`
+    margin-bottom: 32px;
+  `}
+`
+
+const PaginationContainer = styled.div`
+  margin-bottom: 120px;
+  ${mq.mobileOnly`
+    margin-bottom: 64px;
+  `}
+`
+
+const SnackBarContainer = styled.div`
+  z-index: 1;
+  position: fixed;
+  transition: opacity 100ms;
+  opacity: ${props => (props.showSnackBar ? 1 : 0)};
+  max-width: 440px;
+  width: 100%;
+  left: 50%;
+  ${mq.desktopAndAbove`
+    bottom: 24px;
+    transform: translateX(-50%);
+  `}
+  ${mq.tabletAndBelow`
+    bottom: calc(env(safe-area-inset-bottom, 0) + 60px + 8px);
+    transform: translateX(-50%);
+    padding: 0 16px;
+  `}
+`
+
+const SnackBarDiv = styled.div`
+  display: flex;
+  justify-content: center;
+`
+
+const EmptyStateConatiner = styled.div`
+  margin-top: 72px;
+  margin-bottom: 120px;
+`
+
+const SavedBookmarks = ({
+  isDeleting,
+  isFetching,
+  jwt,
+  userID,
+  getMultipleBookmarks,
+  page,
+  totalPages,
+}) => {
+  const [totalSavedBookmarkCount, setTotalSavedBookmarkCount] = useState(0)
+  const [bookmarks, setBookmarks] = useState([])
+  const [showEmptyState, setShowEmptyState] = useState(false)
+  const { releaseBranch } = useContext(CoreContext)
+  const store = useStore()
+  const { removeAction } = useBookmark(store)
+  const { showSnackBar, snackBarText, toastr } = useSnackBar()
+  const navigate = useHistory()
+
+  const removeBookmark = bookmarkID => {
+    removeAction(bookmarkID)
+      .then(() => {
+        toastr({ text: '已取消收藏' })
+        setBookmarks(prevBookmarks =>
+          _.filter(prevBookmarks, bookmark => bookmark.id !== bookmarkID)
```

Comment:
(nice to have) we could use [`remove`](https://lodash.com/docs/4.17.15#remove) to make it easier to understand

Reply:
fixed

Reply:
fixed